src/main/groovy/org/labkey/gradle/task/CreateModule.groovy

```
@@ -0,0 +1,169 @@
+                    addProperty: "new_moduleDestination"
+            )
+            moduleDestination = ant.new_moduleDestination
+
+            project.ant.input(
+                    message: "\nWill this module create and manage a database schema? (y/N)",
+                    addProperty: "new_hasManagedSchema"
+            )
+            hasManagedSchema = !(ant.new_hasManagedSchema.toLowerCase().equals("n"))
+
+            project.ant.input(
+                    message: "\nCreate test stubs (y/N)",
+                    addProperty: "new_createTestFiles"
+            )
+            createTestFiles = ant.new_createTestFiles.toLowerCase().equals("y")
+
+            project.ant.input(
+                    message: "\nCreate API stubs (y/N)",
+                    addProperty: "new_createApiFiles"
+            )
+            createApiFiles = ant.new_createApiFiles.toLowerCase().equals("y")
+        }
+        else {
+            if (project.hasProperty('moduleName')) {
+                moduleName = project.moduleName
+            }
+            if (project.hasProperty('moduleDestination')) {
+                moduleDestination = project.moduleDestination
+            }
+            if (project.hasProperty('hasManagedSchema')) {
+                hasManagedSchema = !(project.hasManagedSchema.toLowerCase().equals("n"))
+            }
+            if (project.hasProperty('createTestFiles')) {
+                createTestFiles = project.createTestFiles.toLowerCase().equals("y")
+            }
+            if (project.hasProperty('createApiFiles')) {
+                createApiFiles = project.createApiFiles.toLowerCase().equals("y")
+            }
+        }
+
+        if (moduleName == null) {
+            project.logger.error("moduleName is not specified")
+        }
+        if (moduleDestination == null) {
+            project.logger.error("moduleDestination is not specified")
+        }
```

We should probably throw an exception if one of these is null since we can't create a module without these.